File: product/src/main/java/com/lexuancong/product/viewmodel/product/ProductFeaturePagingVm.java
package com.lexuancong.product.viewmodel.product;

import java.util.List;

public record ProductFeaturePagingVm(
        List<ProductPreviewVm> productPreviewsPayload,
        int pageIndex,
        int pageSize,
        int totalElements,
        int totalPages,
        boolean isLast
) {

}

File: nextjs-frontend_store/models/product/productFeaturePagingVm.ts
import {ProductPreviewVm} from "@/models/product/ProductPreviewVm";

export type ProductFeaturePagingVm = {
    productPreviewsPayload: ProductPreviewVm[],
    pageIndex: number,
    pageSize: number,
    totalElements: number,
    totalPages: number,
    isLast : boolean


}